torchsparse/utils/kernel_region.py

```diff
@@ -20,31 +20,55 @@ def __init__(self,
         self.tensor_stride = tensor_stride
         self.dilation = dilation
 
-        if kernel_size % 2 == 0:
-            # even
-            region_type = 0
-        else:
-            # odd
-            region_type = 1
+        if not isinstance(kernel_size, (list, tuple)):
+            if kernel_size % 2 == 0:
+                # even
+                region_type = 0
+            else:
+                # odd
+                region_type = 1
 
-        self.region_type = region_type
+            self.region_type = region_type
 
-        single_offset = (
-            np.arange(-kernel_size // 2 + 1, kernel_size // 2 + 1) *
-            tensor_stride * dilation).tolist()
+            single_offset = (
+                np.arange(-kernel_size // 2 + 1, kernel_size // 2 + 1) *
+                tensor_stride * dilation).tolist()
 
-        x_offset = single_offset if 0 in dim else [0]
-        y_offset = single_offset if 1 in dim else [0]
-        z_offset = single_offset if 2 in dim else [0]
+            x_offset = single_offset if 0 in dim else [0]
+            y_offset = single_offset if 1 in dim else [0]
+            z_offset = single_offset if 2 in dim else [0]
 
-        if self.region_type == 1:
-            kernel_offset = [[x, y, z] for z in z_offset for y in y_offset
-                             for x in x_offset]
+            if self.region_type == 1:
+                kernel_offset = [[x, y, z] for z in z_offset for y in y_offset
+                                for x in x_offset]
+            else:
+                kernel_offset = [[x, y, z] for x in x_offset for y in y_offset
+                                for z in z_offset]
+            kernel_offset = np.array(kernel_offset)
+            self.kernel_offset = torch.from_numpy(kernel_offset).int()
         else:
-            kernel_offset = [[x, y, z] for x in x_offset for y in y_offset
-                             for z in z_offset]
-        kernel_offset = np.array(kernel_offset)
-        self.kernel_offset = torch.from_numpy(kernel_offset).int()
+            if dim == [0,1,2] and len(kernel_size) ==3:            
+                kernel_x_size = kernel_size[0] if 0 in dim else 1
```

Comment:
kernel_x_size = kernel_size[0]
and below

Reply:
Done